### ISSUE TITLE
grpc-bom to specify type:pom for protoc-gen-grpc-java

### ISSUE THREAD

**Issue by suztomo** [grpc/grpc-java#6219]
### Is your feature request related to a problem? Please describe.

When our tool ([cloud-opensource-java dashboard](https://github.com/GoogleCloudPlatform/cloud-opensource-java/#google-cloud-platform-java-dependency-dashboard)) tries to read grpc-bom to check the compatibility of Google Java libraries, it fails because our tool tries to download a "JAR" file for protoc-gen-grpc-java (which has type:pom).  https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/943

While [`io.grpc:protoc-gen-grpc-java:1.24.0`](https://search.maven.org/artifact/io.grpc/protoc-gen-grpc-java/1.24.0/pom) properly declares type:pom, [grpc-bom](https://search.maven.org/artifact/io.grpc/grpc-bom/1.24.0/pom) contains:

```
      <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>protoc-gen-grpc-java</artifactId>
        <version>1.24.0</version>
      </dependency>
```

From gRPC BOM reader's perspective, I cannot tell the packaging as pom.

### Describe the solution you'd like

I want grpc-bom to declare "type: pom" for `protoc-gen-grpc-java`. I guess the change would be at https://github.com/grpc/grpc-java/blob/master/bom/build.gradle#L32

### Describe alternatives you've considered

Alternatively, I can implement special case for protoc-gen-grpc-java in our tool. But declaring the packaging in the BOM properly would make more sense.

### Additional context
Add any other context about the feature request here.

**Comment by suztomo** [comment]
@creamsoup Thank you for quick turnaround.

**Comment by suztomo** [comment]
I see this is available in [1.24.1](https://search.maven.org/artifact/io.grpc/grpc-bom/1.24.1/pom):

```
      <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>protoc-gen-grpc-java</artifactId>
        <version>1.24.1</version>
        <type>pom</type>
      </dependency>
```

Nice.

**Comment by creamsoup** [comment]
I am surprised that you noticed the patch release even though we haven't announced.
